sklearn/naive_bayes.py

```diff
@@ -522,7 +523,7 @@ def fit(self, X, y, sample_weight=None):
 
         # convert to float to support sample weight consistently;
         # this means we also don't have to cast X to floating point
-        Y = Y.astype(np.float64)
+        Y = astype(Y, np.float64, copy=False)
```

Comment:
Please revert that change and maybe improve the inline comment to mention that as of now `LabelBinarizer`'s `fit_transform` method returns arrays with `dtype=np.int64`